src/Sfa.Tl.Matching.Web/Controllers/ProviderController.cs

```diff
@@ -120,29 +126,50 @@ private async Task<SearchViewModel> GetSearchResults(SearchParametersViewModel v
         {
             _logger.LogInformation($"Searching for route id {viewModel.SelectedRouteId}, postcode {viewModel.Postcode}");
 
+            var searchViewModel = await SearchProviders(viewModel.Postcode, viewModel.SelectedRouteId, viewModel.SearchRadius);
+            
+            return searchViewModel;
+        }
+
+        private async Task<SearchViewModel> SearchProviders(string postcode, int? routeId, int searchRadius)
+        {
             var searchResults = await _providerService.SearchProvidersByPostcodeProximity(new ProviderSearchParametersDto
             {
-                Postcode = viewModel.Postcode,
-                SelectedRouteId = viewModel.SelectedRouteId,
-                SearchRadius = viewModel.SearchRadius
-            } );
+                Postcode = postcode,
+                SelectedRouteId = routeId,
+                SearchRadius = searchRadius
+            });
 
-            var resultsViewModel = new SearchViewModel
+            var viewModel = new SearchViewModel
             {
-                SearchResults = new SearchResultsViewModel
-                {
-                    Results = _mapper.Map<List<SearchResultsViewModelItem>>(searchResults)
-                },
                 SearchParameters = new SearchParametersViewModel
                 {
+                    SearchRadius = searchRadius,
+                    Postcode = postcode,
+                    SelectedRouteId = routeId,
                     RoutesSelectList = _mapper.Map<SelectListItem[]>(GetRoutes()),
-                    SearchRadius = viewModel.SearchRadius,
-                    SelectedRouteId = viewModel.SelectedRouteId,
-                    Postcode = viewModel.Postcode?.Trim()
+                },
+                SearchResults = new SearchResultsViewModel
+                {
+                    Results = _mapper.Map<List<SearchResultsViewModelItem>>(searchResults)
                 }
             };
 
-            return resultsViewModel;
+            return viewModel;
+        }
+
+        [HttpPost]
+        [Route("provider-results-within-{SearchRadius}-miles-of-{Postcode}-for-route-{SelectedRouteId}", Name = "CreateReferral_Post")]
+        public async Task<IActionResult> CreateReferral(CreateReferralViewModel viewModel)
```

Comment:
No point in having fancy route Url because its a post
and Rename it to something like ValidateIfAtLeastOneProviderIsSelectedfromSearchResult
this action is clearly not creating Referral